lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -1011,11 +1011,13 @@ def test_lines_dists():
 def test_lines_dists_nowarning():
     # Smoke test to see that no RuntimeWarning is emitted when two first
     # arguments are the same, see GH#22624
-    p0 = (10, 30)
-    p1 = (20, 150)
-    proj3d._line2d_seg_dist(p0, p0, p1)
+    p0 = (10, 30, 50)
+    p1 = (10, 30, 20)
+    p2 = (20, 150)
+    proj3d._line2d_seg_dist(p0, p0, p2)
+    proj3d._line2d_seg_dist(p0, p1, p2)
```

Comment:
Um, this now accepts a 3d line and a 2d point? I'm not clear what the semantics here is or whether this is desired. At the very least we should document the expected input dimensions for the function.

Reply:
This is an internal helper that I think we pulled out.

Given that @scottshambaugh was seeing this warning we are apparently passing in 3D points in practice.

While this is an odd internal wart, I do not think it is worth the effort we have collectively thrown at this function recently!

Reply:
My point is that we are doing some ill-defined things here. Note that this function is only used in `format_coord` and I wouldn't be surprised if it's the cause of  #22775.

The earlier work #22628 was defensive in only bailing out when both line-defining points were equal in 3d. That does not alter the behavior.

I don't know what the correct behavior here should be, but I believe that neglecting the z-direction will add another quirk to the coord determination.

More generally, I support @anntzer's [view](https://github.com/matplotlib/matplotlib/issues/22624#issuecomment-1063945593) that we should stop displaying 3D coords, because we can't get them from a 2d pointer consistently.